### PR TITLE
Display adresses on map

### LINKED ISSUE #8: Intégrer les données à la carte sous forme de points
- [x] Toutes les lignes du csv qui ont des coordonnées gps ont un point correspondant sur la carte

### PR DESCRIPTION
**Pull request by clairezed**
closes #8

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -15,10 +15,11 @@
...
         crossorigin=""/>
         <link rel="stylesheet" href="style.css" />
+
         <script crossorigin="anonymous" src="https://cdn.polyfill.io/v2/polyfill.min.js?features=all"></script>
-        <script src="https://unpkg.com/leaflet@1.3.1/dist/leaflet.js"
-        integrity="sha512-/Nsx9X4HebavoBvEBuyp3I7od5tA0UzAxs+j83KgC8PU0kgB4XiK4Lfe4y4cgBtaRJQEIFCW+oC506aPT2L1zw=="
-        crossorigin="anonymous"></script>
+        <script crossorigin="anonymous" src="https://unpkg.com/leaflet@1.3.1/dist/leaflet.js"
+        integrity="sha512-/Nsx9X4HebavoBvEBuyp3I7od5tA0UzAxs+j83KgC8PU0kgB4XiK4Lfe4y4cgBtaRJQEIFCW+oC506aPT2L1zw=="></script>
+        <script crossorigin="anonymous" src='https://cdn.jsdelivr.net/npm/papaparse@4.5.0/papaparse.min.js'></script>
```
Comment (DavidBruant) [inline index.html:22]: Interesting, i didn't know about this parser
My usual choice for a client-side csv parser is [d3-fetch csv function](https://github.com/d3/d3-fetch#csv), but this works too

File: index.js [hunk 2]
```diff
@@ -1,8 +1,30 @@
-var mymap = L.map('mapid').setView([48.1833300,  6.4500000], 13);
+const guideMap = L.map('mapid', {
```
Comment (DavidBruant) [inline index.js]: Nice rename!

File: index.js [hunk 3]
```diff
@@ -1,8 +1,30 @@
...
     maxZoom: 18,
     id: 'mapbox.streets',
     accessToken: 'your.mapbox.access.token'
-}).addTo(mymap);
\ No newline at end of file
+}).addTo(guideMap);
+
+Papa.parse('guide-conso-data.csv', {
+    header: true,
+	download: true,
```
Comment (DavidBruant) [inline index.js]: The indent looks weird here

File: index.js [hunk 4]
```diff
@@ -1,8 +1,30 @@
...
+	complete: function(results) {
+		displayAddressesOnMap(results.data);
+    },
+    error: function(error, file) {
+        console.log(error);
+    }
+});
+
+function displayAddressesOnMap(addresses) {
+    addresses.map(address => {
```
Comment (DavidBruant) [inline index.js]: `.map` does what is intended, but is usually used to create another array
If you want to only iterate over the array, you may prefer `.forEach`. A [for-of loop](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of#Examples) works too

File: index.js [hunk 5]
```diff
@@ -1,8 +1,30 @@
...
+        let lat = address['Latitude'];
+        let long = address['Longitude'];
```
Comment (DavidBruant) [inline index.js]: if you don't intend to change the variable value, prefer `const` to `let`

Comment (DavidBruant) [reply index.js]: Another option here is to use [destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Object_destructuring)
```js
const {Latitude: lat, Longitude : long} = address;
```

File: index.js [hunk 6]
```diff
@@ -1,8 +1,30 @@
...
+}).addTo(guideMap);
+
+Papa.parse('guide-conso-data.csv', {
+    header: true,
+	download: true,
+	complete: function(results) {
+		displayAddressesOnMap(results.data);
+    },
+    error: function(error, file) {
+        console.log(error);
```
Comment (DavidBruant) [inline index.js]: The [console object](https://developer.mozilla.org/en-US/docs/Web/API/Console) has lots of methods including one dedicated to showing a message as an error in the console : `console.error`

File: index.js [hunk 7]
```diff
@@ -1,8 +1,30 @@
...
+const guideMap = L.map('mapid', {
+    center: [48.1833300,  6.4500000],
+    zoom: 11
+});
 
 L.tileLayer('https://api.tiles.mapbox.com/v4/{id}/{z}/{x}/{y}.png?access_token=pk.eyJ1IjoiY2xhaXJlemVkIiwiYSI6ImNqMWwydTFwcTAwMDgzM253Njg1aTY1c2UifQ.77uslVJ8ewCRqXqyH6J43w', {
     attribution: 'Map data &copy; <a href="https://www.openstreetmap.org/">OpenStreetMap</a> contributors, <a href="https://creativecommons.org/licenses/by-sa/2.0/">CC-BY-SA</a>, Imagery © <a href="https://www.mapbox.com/">Mapbox</a>',
     maxZoom: 18,
     id: 'mapbox.streets',
     accessToken: 'your.mapbox.access.token'
```
Comment (DavidBruant) [inline index.js]: The token is provided in the url above, so this line can be removed i think
(sorry i didn't catch it earlier)

**Review by DavidBruant — CHANGES_REQUESTED** [review]
Good overall
I'm marking request changes for the `.map`. The rest is more or less cosmetics

File: index.js (hunk 5 above)
Comment (clairezed) [reply index.js]: yeah ! yeah ! Have never used destructuring yet ! Let's do that !

**Comment by clairezed** [comment]
Je te laisse jeter un oeil et merger si ça te convient ?

File: index.html [hunk 8]
```diff
@@ -15,13 +15,14 @@
...
         <script defer src="index.js"></script>
     </head>
     <body>
-        <div id="mapid"></div>
+        <div id="mapid" class="map-container"></div>
```
Comment (DavidBruant) [inline index.html]: You don't need the id if you have the class

File: index.js [hunk 9]
```diff
@@ -1,8 +1,28 @@
-var mymap = L.map('mapid').setView([48.1833300,  6.4500000], 13);
+const guideMap = L.map('mapid', {
```
Comment (DavidBruant) [inline index.js]: instead of the id (because you'll remove it), you can pass the DOM element directly

**Comment by DavidBruant** [comment]
LGTM modulo l'histoire de l'id

**Comment by clairezed** [comment]
Damned, I've been writing French in my last comment, sorry for that. 

I've replaced map `id` with `data-attribute`, so as to separate style with js selector.

File: index.js [hunk 10]
```diff
@@ -1,8 +1,29 @@
-var mymap = L.map('mapid').setView([48.1833300,  6.4500000], 13);
+const mapElement = document.querySelector('[data-map="main"]')
```
Comment (DavidBruant) [inline index.js:1]: pourquoi pas `document.querySelector('.map-container')` ?

**Comment by DavidBruant** [comment]
oh i've been writing french as well, sorry :-/
i'm having a hard time switching back and forth between French and English in different repos
Is it ok for you if we write the language we choose in the infusion file?

File: index.js (hunk 10 above)
Comment (clairezed) [reply index.js:1]: To separate css style with js selector. I took this habit in my previous agency, it saved our lives a few time : style may change, css classes may change, and it shouldn't impact js.

Comment (DavidBruant) [reply index.js:1]: interesting... i'm not sure i agree, but that's fine in our case
PR can be merged :-)

**Review by DavidBruant — APPROVED** [review]

**Pull request merged**